src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,249 @@
+            }
+
+            // zip up and email the csv file
+            AppInfo appInfo = dynamoHelper.getApp(appId);
+            AccountInfo accountInfo = bridgeHelper.getAccountInfo(appId, userId);;
+            zipFile = fileHelper.newFile(tmpDir, ZIP_FILE_NAME);
+
+            zipFiles(csvFile, zipFile, password);
+            sesHelper.sendEmailWithAttachmentToAccount(appInfo, accountInfo, zipFile.getAbsolutePath());
+
+        } catch (BridgeSDKException ex) {
+            int status = ex.getStatusCode();
+            if (status >= 400 && status < 500) {
+                throw new PollSqsWorkerBadRequestException(ex);
+            } else {
+                throw new RuntimeException(ex);
+            }
+        } finally {
+            cleanupFiles(csvFile, zipFile, tmpDir);
+            LOG.info("request took " + requestStopwatch.elapsed(TimeUnit.SECONDS) +
+                    " seconds for userId =" + userId + ", app=" + appId);
+        }
+    }
+
+    /** Write all of the account summaries to the CSV file */
+    private void writeAccountSummaries(CSVWriter csvFileWriter, List<AccountSummary> accountSummaries, int offsetBy,
+                                       String appId, String orgMembership, String studyId) throws IOException, InterruptedException {
+        // write csv headers
+        String[] csvHeaders = getCsvHeaders(accountSummaries.get(0));
+        csvFileWriter.writeNext(csvHeaders);
+
+        // write the rest of the account summaries
+        while (!accountSummaries.isEmpty()) {
+            for (AccountSummary accountSummary : accountSummaries) {
+                String[] row = getAccountSummaryArray(accountSummary, csvHeaders);
+                csvFileWriter.writeNext(row);
+            }
+            // get next set of account summaries
+            offsetBy += PAGE_SIZE;
+            accountSummaries = bridgeHelper.getAccountSummariesForApp(appId, orgMembership, offsetBy, 0, studyId);
+
+            // avoid burning out Bridge Server
+            Thread.sleep(THREAD_SLEEP_INTERVAL);
+        }
+    }
+
+    /** Given an Object, return a String array of all of its attribute keys
+     *  Visible for unit testing
+     */
+    String[] getCsvHeaders(Object accountSummary) {
+        JsonElement json = RestUtils.toJSON(accountSummary);
+        Set<String> keys = json.getAsJsonObject().keySet();
+        List<String> list = new ArrayList<>(keys);
+        return list.toArray(new String[0]);
+    }
+
+    /** Given an AccountSummary, return a String array of all of its attribute values
+     *  Visible for unit testing
+     */
+    String[] getAccountSummaryArray(AccountSummary accountSummary, String[] csvHeaders) {
+        JsonObject json = RestUtils.toJSON(accountSummary).getAsJsonObject();
+        ArrayList<String> list = new ArrayList<>();
+
+        for (int i = 0; i < csvHeaders.length; i++) {
+            String value = json.get(csvHeaders[i]) == null ? "" : json.get(csvHeaders[i]).toString();
```

Comment:
If the value is a string type, we should do .textValue() instead of .toString(), so we don't end up with a bunch of quotes everywhere.

Reply:
Oh yes I see your point!! .textValue() doesn't come up, but .getAsString() does and it seems to work after testing a a couple of times. Please let me know if that's okay with you.

Reply:
I'm running into a problem after changing it to .getAsString(). When I call json.get(csvHeaders[i]), it will sometimes return a JsonElement, sometimes return a JsonPrimitive, and also JsonObject. The issue with when it returns a JsonObject is that .getAsString() doesn't work for it, and only .toString() does. I'll try to dig deeper..

Reply:
I got GSON and Jackson confused. For GSON, you'll want to check .isPrimitive() to be able to use .getAsString()